client/orderbook/orderbook.go

```diff
@@ -604,3 +613,40 @@ func (ob *OrderBook) BestFill(sell bool, qty uint64) ([]*Fill, bool) {
 func (ob *OrderBook) BestFillMarketBuy(qty, lotSize uint64) ([]*Fill, bool) {
 	return ob.sells.bestFill(qty, true, lotSize)
 }
+
+// SetMatchesSummary set matches summary. If the matches summary length grows
+// bigger than 100, it will slice out the ones first added.
+func (ob *OrderBook) SetMatchesSummary(matches map[uint64]uint64, ts uint64) {
+	newMatchesSummary := make([]*MatchSummary, len(matches))
+	i := 0
+	for rate, qty := range matches {
+		newMatchesSummary[i] = &MatchSummary{
+			Rate: rate,
+			Qty:  qty,
+			Age:  ts,
+		}
+		i++
+	}
+
+	ob.matchSummaryMtx.Lock()
+	defer ob.matchSummaryMtx.Unlock()
+	if ob.matchesSummary == nil {
+		ob.matchesSummary = newMatchesSummary
+		return
+	}
+	ob.matchesSummary = append(newMatchesSummary, ob.matchesSummary...)
+	maxLength := 1000
+	// if maxLength is greater than 100, slice it and left the first 100.
+	if len(ob.matchesSummary) > maxLength+100 {
+		ob.matchesSummary = ob.matchesSummary[maxLength:]
+	}
+}
+
+func (ob *OrderBook) GetMatchesSummary() []*MatchSummary {
+	ob.matchSummaryMtx.Lock()
+	if ob.matchesSummary == nil {
+		ob.matchesSummary = make([]*MatchSummary, 0)
+	}
+	ob.matchSummaryMtx.Unlock()
+	return ob.matchesSummary
+}
```

Comment:
I don't see a great reason to ensure we are returning a non nil slice. Caller should handle it properly.